### PR TITLE
Update lightship types to 6.1

### PR DESCRIPTION
**Pull request by janslow**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  - https://github.com/gajus/lightship#usage
  - [Official Flow Types](https://github.com/gajus/lightship/blob/master/src/types.js)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/lightship/index.d.ts [hunk 1]
```diff
@@ -10,30 +10,57 @@ import { Server } from 'http';
  * A teardown function called when shutdown is initialized.
  * @param userConfiguration User configuration object
  */
-export function createLightship(userConfiguration?: UserConfigurationType): LightshipType;
+export function createLightship(configuration?: ConfigurationInputType): LightshipType;
```
Comment (janslow) [inline types/lightship/index.d.ts:13]: Renaming this to match the official type names in the [Usage instructions](https://github.com/gajus/lightship#usage) and [Flow Types](https://github.com/gajus/lightship/blob/master/src/types.js).

File: types/lightship/index.d.ts [hunk 2]
```diff
@@ -10,30 +10,57 @@ import { Server } from 'http';
...
 export type BeaconContextType = object;
 
 export interface BeaconControllerType {
     die: () => Promise<void>;
 }
 
-export interface UserConfigurationType {
-    /* Run Lightship in local mode when Kubernetes is not detected. Default: true. */
+export interface ConfigurationInputType {
+    /**
```
Comment (janslow) [inline types/lightship/index.d.ts:22]: Change the comments to be in JSDoc/TSDoc format (i.e., `/**`, not `/*`), so that editors display them

**Review by karoun — APPROVED** [review]
Nice 👍

**Comment by janslow** [comment]
Ready to merge

Thanks for the quick review @karoun!

**Pull request merged**